scalalib/src/JavaModule.scala

```diff
@@ -187,12 +187,11 @@ trait JavaModule extends mill.Module
     * All individual source files fed into the Java compiler
     */
   def allSourceFiles = T{
-    def isHiddenFile(path: os.Path) = path.last.startsWith(".")
     for {
       root <- allSources()
       if os.exists(root.path)
-      path <- (if (os.isDir(root.path)) os.walk(root.path) else Seq(root.path))
-      if os.isFile(path) && ((path.ext == "java") && !isHiddenFile(path))
+      path <- if (os.isDir(root.path)) os.walk(root.path) else Seq(root.path)
+      if os.isFile(path) && ((path.ext == "java") && !path.toIO.isHidden)
```

Comment:
Why is that change?

Reply:
The `isHidden` is already defined in Java IO and is probably more robust wrt cross platform (i.e on windows a hidden file don't start with a `.`)

Reply:
Yeah, but I think a dot file should always be excluded and treated not as sourcefile. If you make that logic OS-specific, your build end up to be non-reproducible.

Reply:
Ah, I see. We want a file starting with a `.` to always  be excluded no matter what OS it it.
Reverting this change.